rir/src/compiler/opt/constantfold.cpp

```diff
@@ -78,10 +80,23 @@ void Constantfold::apply(RirCompiler& cmp, Closure* function,
             FOLD(Eq, symbol::Eq);
             FOLD(Neq, symbol::Ne);
             FOLD(Pow, symbol::Pow);
+
+            if (auto assume = Assume::Cast(i)) {
+                auto condition = assume->arg<0>().val();
+                if (auto isObj = IsObject::Cast(condition))
+                    if (!isObj->arg<0>().val()->type.maybeObj())
+                        next = bb->remove(ip);
+
+                FOLD2(condition, Identical,
```

Comment:
ultimatively the issue is that we do not have a native true/false value in pir atm. So native booleans are only used in branches, so instead of first constant folding tests to true/false and then remove branches on true/false, i do both at the same time.

Reply:
I am gonna put a todo there...